fmriprep/cli/parser.py

```diff
@@ -62,6 +62,18 @@ def __call__(self, parser, namespace, values, option_string=None):
             print(msg, file=sys.stderr)
             delattr(namespace, self.dest)
 
+    class ToDict(Action):
+        def __call__(self, parser, namespace, values, option_string=None):
+            d = {}
+            for i_kv, kv in enumerate(values):
+                if '=' not in kv:
+                    k = f'deriv-{i_kv}'
+                    v = kv
+                else:
+                    k, v = kv.split('=')
+                d[k] = Path(v)
```

Comment:
Not a huge fan of `deriv-[0..n]`. What if we take the basename by default? And then raise an error if there are multiple with the same name (whether passed explicitly or not)?

```suggestion
            d = {}
            for spec in values:
                try:
                    name, loc = spec.split('=')
                    loc = Path(loc)
                except ValueError:
                    loc = Path(spec)
                    name = loc.name

                if name in d:
                    raise ValueError(f"Received duplicate derivative name: {name}")
                d[k] = Path(v)
```

Reply:
I like it. I can't think of any reasonable cases where there would be duplicate derivative folder names, so I don't think there's a problem with automatically inferring the name.